src/controllers/admin/external-user-controller.ts

```diff
@@ -4,12 +4,27 @@ import { createExternalUserSchema } from "../../utils/validation/external-user-s
 import * as ExternalUserService from "../../services/external-user-service"
 
 /**
- * List external users.
+ * List external users based on provided filters.
+ * @param req.query.name - Filter by name.
+ * @param req.query.company - Filter by company.
+ * @param req.query.role - Filter by role.
+ * @param req.query.page - Page number for pagination.
  */
 export const index = async (req: Request, res: Response) => {
   try {
-    const externalUsers = await ExternalUserService.getAll()
-    res.json(externalUsers)
+    const { name, company, role, page } = req.query
+
+    const externalUsers = await ExternalUserService.getAllByFilters(
+      name as string,
+      company as string,
+      role as string,
+      page as string
+    )
+
+    res.json({
```

Comment:
I think you can do `res.json(externalUsers)` here 🤔

Reply:
Oh my bad, updated it already tysm!